### PR TITLE
Feature/44 payment

### PR DESCRIPTION
**Pull request by fasbrothers**
- Change payment design and add accommodation details
- Add infinite scroll on the accommodation owners' page
- Resolve booking

### CODE REVIEW

File: src/api/mutations/booking/useBookingRoom.ts [hunk 1]
```diff
@@ -17,11 +16,10 @@ export const useBookingRoom = () => {
       accommodationId: string;
     }) => {
       const { data } = await httpClient.post<BookResponse>(ENDPOINTS.booking.book, reservationData);
-      return data;
+      return data.data;
     },
     onSuccess: (data) => {
```
Comment (gitkoshelew) [inline src/api/mutations/booking/useBookingRoom.ts:23]: spread 'data' here

File: src/pages/payment/Payment.tsx [hunk 2]
```diff
@@ -39,6 +69,17 @@ const Payment = () => {
     [mutate]
   );
 
+  const calculateTotalPrice = useCallback(
```
Comment (gitkoshelew) [inline src/pages/payment/Payment.tsx]: why do you use useCallback here? you dont pass the function as property. here should be useMemo and accomodation.price should be one of dependencies

File: src/pages/payment/Payment.tsx [hunk 3]
```diff
@@ -52,31 +93,62 @@ const Payment = () => {
...
+                alt={accommodation.title}
+                onError={handleErrorInImage}
+              />
+            </Box>
+            <Stack justifyContent={'inherit'} sx={styles.payment_accommodation_details}>
+              <Typography fontWeight={600} variant={mobileScreen ? 'sm' : 'subtitle1'}>
+                {accommodation.title}
+              </Typography>
+              <Typography variant="sm">
+                {truncateReview(accommodation.description, mobileScreen ? 20 : 80)}
```
Comment (gitkoshelew) [inline src/pages/payment/Payment.tsx]: spread accomodation above return

File: src/pages/payment/Payment.tsx [hunk 4]
```diff
@@ -52,31 +93,62 @@ const Payment = () => {
...
-      {isPending ? <LoadingPrimary height="10vh" /> : paymentMethods[paymentOption]}
-    </Box>
+    <>
+      <Typography variant="lg">Payment</Typography>
+      <Stack sx={styles.payment_container}>
+        <Box sx={styles.payment_form}>
+          <FormControl>
+            <FormLabel>Payment Option</FormLabel>
+            <RadioGroup row value={paymentOption} onChange={handlePaymentChange}>
+              <FormControlLabel value={PAYMENT_OPTION.card} control={<Radio />} label="Card" />
```
Comment (gitkoshelew) [inline src/pages/payment/Payment.tsx]: add translations, please

**Review by gitkoshelew — APPROVED** [review]

**Review by MdRasulov — APPROVED** [review]

**Pull request merged**